### DISCUSSION TITLE
Peacock segmentation fault problem

### DISCUSSION THREAD

**Discussion by Balbest** [idaholab/moose#27566] [Q&A General]
When I try to run peacock (from peacock env), the following error is emerging: Segmentation fault (core dumped).
Paraview runs with no problems, libmesh is updated, chigger/tests/window/window_clear.py returns the same segmentation fault.
For some reason, I had some problems with environment variables (i.e. not set MOOSE_DIR). 
<details><summary>test_PeacockMainWindow.py returns:
</summary>
<p>
testConnections (__main__.Tests) ... QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 180, 180, 180, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 180, 180, 180, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
No executable found for method type(s): opt, dbg, oprof, devel
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 180, 180, 180, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 180, 180, 180, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
WARNING
 '/home/balbest/Workmain/moose_projects/moose/test/moose_test-opt -options_left 0 --json' exited with non zero status 127.

Please make sure your application is built and able to execute the given arguments.
Working dir: /home/balbest/Workmain/moose_projects/moose/python/peacock/tests/peacock_app/PeacockMainWindow
Output: /home/balbest/Workmain/moose_projects/moose/test/moose_test-opt: error while loading shared libraries: libmoose-opt.so.0: cannot open shared object file: No such file or directory

WARNING
 Failed to load json from '/home/balbest/Workmain/moose_projects/moose/test/moose_test-opt': '/home/balbest/Workmain/moose_projects/moose/test/moose_test-opt -options_left 0 --json' exited with non zero status 127.

Please make sure your application is built and able to execute the given arguments.
Working dir: /home/balbest/Workmain/moose_projects/moose/python/peacock/tests/peacock_app/PeacockMainWindow
Output: /home/balbest/Workmain/moose_projects/moose/test/moose_test-opt: error while loading shared libraries: libmoose-opt.so.0: cannot open shared object file: No such file or directory

FAIL
testConsole (__main__.Tests) ... QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 180, 180, 180, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 180, 180, 180, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
No executable found for method type(s): opt, dbg, oprof, devel
ok

======================================================================
FAIL: testConnections (__main__.Tests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/balbest/Workmain/moose_projects/moose/python/peacock/tests/peacock_app/PeacockMainWindow/./test_PeacockMainWindow.py", line 39, in testConnections
    self.assertIn(path, w.windowTitle())
AssertionError: '/home/balbest/Workmain/moose_projects/moose/test/moose_test-opt' not found in 'Peacock'

----------------------------------------------------------------------
Ran 2 tests in 0.486s

FAILED (failures=1)

</p>
</details> 

moose_test-opt itself runs as usual

**Comment by GiudGiud** [comment]
Hello

Which OS are you using?

Guillaume

**Reply by Balbest** [reply]
Ubuntu 22

**Reply by GiudGiud** [reply]
can you set the MOOSE_DIR environment variable, compile moose if you have not, and try again to see if those warnings about finding moose disappear ?

**Reply by Balbest** [reply]
okay, after I repeated the process, I've got exactly the same output described here: https://github.com/idaholab/moose/discussions/26884#discussioncomment-8625222
And I also don't have any other VTK installations.
Repeating this https://github.com/idaholab/moose/discussions/26884#discussioncomment-8644130 
<details><summary>returns the following:</summary>
<p>

(/home/balbest/home/balbest/Temp_conda/_env/.envs/peacock-testing) bash-5.1$ ./peacock
Traceback (most recent call last):
  File "/home/balbest/Workmain/moose_projects/moose/python/peacock/./peacock", line 36, in <module>
    run_peacock()
  File "/home/balbest/Workmain/moose_projects/moose/python/peacock/./peacock", line 28, in run_peacock
    sys.exit(main(sys.argv))
  File "/home/balbest/Workmain/moose_projects/moose/python/peacock/./peacock", line 12, in main
    from peacock.CheckRequirements import has_requirements
  File "/home/balbest/Workmain/moose_projects/moose/python/peacock/__init__.py", line 11, in <module>
    from . import base
  File "/home/balbest/Workmain/moose_projects/moose/python/peacock/base/__init__.py", line 10, in <module>
    from .MooseWidget import MooseWidget
  File "/home/balbest/Workmain/moose_projects/moose/python/peacock/base/MooseWidget.py", line 11, in <module>
    import mooseutils
  File "/home/balbest/Workmain/moose_projects/moose/python/mooseutils/__init__.py", line 23, in <module>
    from .civet_results import get_civet_results, get_civet_hashes
  File "/home/balbest/Workmain/moose_projects/moose/python/mooseutils/civet_results.py", line 24, in <module>
    DEFAULT_JOBS_CACHE = os.path.join(os.getenv('HOME'), '.local', 'share', 'civet', 'jobs')
  File "/home/balbest/home/balbest/Temp_conda/_env/.envs/peacock-testing/lib/python3.10/posixpath.py", line 76, in join
    a = os.fspath(a)
TypeError: expected str, bytes or os.PathLike object, not NoneType


</p>
</details>

**Reply by GiudGiud** [reply]
can you export the HOME environment variable? It seems to fail on that line that does `os.getenv('HOME')`

**Reply by Balbest** [reply]
yes, it was not defined. So now ./peacock returns "Aborted (core dumped)"

**Reply by GiudGiud** [reply]
well nothing I can do from that unfortunately. 

I would recommend:
- try the workbench GUI instead : https://www.ornl.gov/onramp/neams-workbench
- use the input files outside of the GUI and use paraview for visualizing meshes / results. That's how I do all my work

**Reply by Balbest** [reply]
yes, I switch to paraview then, thanks for your efforts!
